service/dap/server.go

```diff
@@ -518,42 +565,68 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 
 	s.setLaunchAttachArgs(request)
 
-	var targetArgs []string
-	args, ok := request.Arguments["args"]
-	if ok {
-		argsParsed, ok := args.([]interface{})
-		if !ok {
-			s.sendErrorResponse(request.Request,
-				FailedToLaunch, "Failed to launch",
-				fmt.Sprintf("'args' attribute '%v' in debug configuration is not an array.", args))
-			return
-		}
-		for _, arg := range argsParsed {
-			argParsed, ok := arg.(string)
-			if !ok {
-				s.sendErrorResponse(request.Request,
-					FailedToLaunch, "Failed to launch",
-					fmt.Sprintf("value '%v' in 'args' attribute in debug configuration is not a string.", arg))
-				return
-			}
-			targetArgs = append(targetArgs, argParsed)
-		}
+	targetArgs, ok := s.parseTargetArgs(request, ok)
+	if !ok {
+		return
 	}
 
 	s.config.ProcessArgs = append([]string{program}, targetArgs...)
 	s.config.Debugger.WorkingDir = filepath.Dir(program)
 
 	// Set the WorkingDir for this program to the one specified in the request arguments.
-	wd, ok := request.Arguments["wd"]
-	if ok {
-		wdParsed, ok := wd.(string)
-		if !ok {
+	if !s.setWorkingDir(request) {
+		return
+	}
+
+	if mode == "record" {
```

Comment:
So, the intent here is to have a command that just records the execution of an executable? I don't think this is very useful, it can be done easily by calling rr with a terminal. I would remove this and add a backend parameter to the launch parameters instead.

Reply:
The intent is a bit further than that, I want to provide the same experience as goland, which implies having a 2 stage mode like them, `record and debug`. Since rr is quirky with output paths and always expects a non-existing directory this would require additional code to generate the new path on the run if the user specifies one, or always default to the rr location and handle that for each trace for the end user

At the current state of the PR I don't see it hard to implement, but it is a matter of how do we want to set up the incoming vs code parameters with the lesser complexity, so that's why I chose to implement just a `record` mode first as a reproducible UI convenience for users (namely, the changes at https://github.com/golang/vscode-go/pull/1268)

cc @polinasok, @hyangah to know what do you think about this. Also @suzmue, who's been on the contract of the debug configuration/adapter

Reply:
Also to add: this is currently only for program execution but as I'm seeing from the code test execution should be just a matter of adding its compilation path like we do on debug mode and continue to the record path

Reply:
AFAICT GoLand does not have a two step record and debug, it records the program and immediately starts a debug session on the recording.

Reply:
Sorry for the late response, what I meant is exactly what you describe, from the UI. For that to be implemented, we would need a separate mode and still, the server-side code would need to handle it so some refactoring should be done first (although I think we could make this work seamlessly)

Reply:
I'm not sure what you mean here. What I was saying is that the record mode should do "record and replay" automatically, like GoLand does. By passing the `rr` backend to `debugger.Launch` this will happen automatically, the user can even stop the recording before it finishes by pressing pause, like in GoLand (this is not a coincidence, btw).

Reply:
Seems this still needs to be resolved. I agree with @aarzilli that we shouldn't need a separate mode for this. Delve, with the RR backend, does the "record and replay" 2-step process automatically behind the scenes.

Reply:
This is still unaddressed.

Reply:
Fixed, I removed the record implementation for now and we shall treat it in a separate PR if anything